hack/update-vendor.sh

```diff
@@ -379,12 +379,8 @@ for repo in $(kube::util::list_staging_repos); do
 done
 
 kube::log::status "vendor: updating BUILD files"
-# Assume that anything imported through vendor doesn't need Bazel to build.
-# Prune out any Bazel build files, since these can break the build due to
-# missing dependencies that aren't included by go mod vendor.
-find vendor/ -type f \
-    \( -name BUILD -o -name BUILD.bazel -o -name WORKSPACE \) \
-    -exec rm -f {} \;
+
+# Prune out any Bazel build files, since these are unnecessary.
```

Comment:
at this point, I'd drop this entirely so we get a vendor dir that matches what `go mod vendor` produces (modulo staging symlinks)

Reply:
We can do that, we also need to alter update-bazel.sh to not touch vendor then.

Reply:
done, there were 3 build files added back by this.